lib/services/install.js

```diff
@@ -0,0 +1,169 @@
+var q = require('q');
+var fs = require('fs');
+var path = require('path');
+var shell = require('shelljs');
+var prompt = require('prompt');
+var colors = require('colors/safe');
+
+/**
+ * Installs instances of PencilBlue.
+ * @constructor
+ */
+var InstallService = function() {};
+
+/**
+ * Installs the application with version and location settings.
+ *
+ * @param  {String} [version]   The version of PencilBlue to install.
+ * @param  {String} [directory] The directory to install PencilBlue to.
+ * @return {Promise} A promise that resolves when installation is complete.
+ */
+InstallService.prototype.installWithSettings = function(version, directory) {
+  var deferred = q.defer();
+  var self = this;
+
+  self.confirmRequirements().then(function() {
+    prompt.message = "PencilBlue".cyan;
+  	prompt.start();
+  	prompt.get(self.getInstallSchema(), function(err, siteSettings) {
+  		if(err) {
+  			deferred.reject(err);
+        return;
+  		}
+
+      self.performInstallation(version, directory, siteSettings).then(function() {
+        deferred.resolve();
+      }, function(err) {
+        deferred.reject(err);
+      });
+    });
+  }, function(err) {
+    deferred.reject(err);
+  });
+
+  return deferred.promise;
+};
+
+/**
+ * Confirms that the system has the requirements for installation.
+ *
+ * @return {Promise} A promise that resolves if the requirements are present.
+ */
+InstallService.prototype.confirmRequirements = function() {
+  var deferred = q.defer();
+
+  if(!shell.which('git')) {
+		deferred.reject('git is required to install PencilBlue.');
+		shell.exit(1);
```

Comment:
Does ```shell.exit``` terminate the CLI?  If yes, shouldn't the top level error handler that calls upon this promise chain decide if it should exit?  Otherwise I don't think there is a point rejecting a promise.